src/expr.cpp

```diff
@@ -7708,6 +7708,63 @@ std::pair<llvm::Constant *, bool> SizeOfExpr::GetConstant(const Type *rtype) con
                                              isNotValidForMultiTargetGlobal);
 }
 
+///////////////////////////////////////////////////////////////////////////
+// AllocaExpr
+
+AllocaExpr::AllocaExpr(Expr *e, SourcePos p) : Expr(p, AllocaExprID), expr(e) {}
+
+llvm::Value *AllocaExpr::GetValue(FunctionEmitContext *ctx) const {
+    ctx->SetDebugPos(pos);
+    if (expr == NULL)
+        return NULL;
+    llvm::Value *llvmValue = expr->GetValue(ctx);
+    if (llvmValue == NULL)
+        return NULL;
+    llvm::Value *resultPtr = ctx->AllocaInst((LLVMTypes::VoidPointerType)->getElementType(), llvmValue, "allocaExpr",
+                                             16, false); // 16 byte stack alignment.
+    return resultPtr;
+}
+
+const Type *AllocaExpr::GetType() const { return PointerType::Void; }
+
+void AllocaExpr::Print() const {
+    printf("AllocaExpr (");
+    if (expr != NULL)
+        expr->Print();
+    const Type *t = expr ? expr->GetType() : NULL;
+    if (t != NULL)
+        printf(" [type %s]", t->GetString().c_str());
+    printf(")");
+    pos.Print();
+}
+
+Expr *AllocaExpr::TypeCheck() {
+    if (expr == NULL) {
+        Error(pos, "\"allocate()\" cannot have a \"NULL\" argument.");
```

Comment:
The following code snippet will trigger this-
void *ptr = alloca(nullptr);

It's not a likely scenario but it is still a user error
That's why I thought it makes sense as a user error message.

But looking at clang implementation, it just gets converted to a alloca for size 0 - https://godbolt.org/z/nrxhK7qq9

Reply:
I don't understand this. `nullptr` is an expression converted to `0`, which means AllocaExpr will have an `expr`, which will be a constant zero. Missing an expression is another story and it seems that it could possibly be only in case of broken compiler logic / something caused an error during AST build and for some reason we didn't crashed (though we should).

Reply:
> It's made more complicated by the fact that we might end up with a null for expr if we cannot evaluate an expr during compile time. So, it's a possibility that it's a compiler error as well

I sounds like we should crash when we can't evaluate the expression, not here.

Reply:
Okay. I changed the check. Now, we check for nullexpr in parser and handle it there.